src/AbstractSQLRules2SQL.ts

```diff
@@ -889,6 +892,17 @@ const typeRules: Dictionary<MatchFn> = {
 		const replacement = TextValue(getAbstractSqlQuery(args, 2), indent);
 		return `REPLACE(${str}, ${find}, ${replacement})`;
 	},
+	ExtractJSONPathAsText: (args, indent) => {
+		checkMinArgs('ExtractJSONPathAsText', args, 1);
+		if (engine !== Engines.postgres) {
+			throw new SyntaxError(
+				'ExtractJSONPathAsText not supported on: ' + engine,
+			);
+		}
+		const column = TextValue(getAbstractSqlQuery(args, 0), indent);
+		const path = args[1] ? getAbstractSqlQuery(args, 1)[1] : '';
+		return `${column}::jsonb #>> '{${path}}'`;
```

Comment:
Same thing here with the casting
```suggestion
		return `${column} #>> '{${path}}'`;
```
and then the caller has the option of
```js
['ExtractJSONPathAsText', ['Cast', x, 'JSON'], path]
```
One notable issue is that since the casts look in sbvr-types for the db type https://github.com/balena-io-modules/abstract-sql-compiler/blob/v7.20.2/src/AbstractSQLRules2SQL.ts#L750-L769 there would currently be issues due to the JSON type using TEXT atm: https://github.com/balena-io-modules/sbvr-types/blob/v3.4.8/src/types/json.ts#L4

Reply:
My initial opinion is that the SBVR type should use proper JSON at some point, and we could leave this as-is for now (expecting actual JSON). Wouldn't really want a workaround for a workaround.